addons/rearm/functions/fnc_rearm.sqf

```diff
@@ -27,11 +27,16 @@ if (isNil "_magazineClass") exitWith {ERROR_1("magazineClass nil",_attachedDummy
 ([_magazineClass] call FUNC(getCaliber)) params ["_cal", "_idx"];
 
 // Get magazines that can be rearmed
-private _needRearmMags = [_target, _magazineClass] call FUNC(getNeedRearmMagazines);
-_needRearmMags params ["_needRearm", "_turretPath", "_cnt", "_pylon"];
+private _needRearmMags = [_target] call FUNC(getNeedRearmMagazines);
+private _needRearmMagsOfClass = _needRearmMags select {(_x select 0) isEqualTo _magazineClass};
 
 // Exit if no magazines need rearming
-if (!_needRearm) exitWith {ERROR_2("Could not find turret for %1 in %2", _magazineClass, typeOf _target);};
+if ((count _needRearmMagsOfClass) == 0) exitWith {ERROR_2("Could not find turret for %1 in %2", _magazineClass, typeOf _target);};
+
+private _currentRearmableMag = _needRearmMagsOfClass select 0;
+private _turretPath = _currentRearmableMag select 1;
+private _pylon = _currentRearmableMag select 3;
+private _magazineCount = _currentRearmableMag select 5;
```

Comment:
In SQF, arrays can be used as lists or pseudo-objects. Pseudo-object is when you use an array to group together multiple related values (like function parameters). There is always a fixed number of elements and they all have a specific purpose. List is when you use an array in the more traditional sense, to hold 0-n values of the same type.

On pseudo-objects, it's clearly better to use `params` to get the individual values. On lists, there rarely is even a need to get the element in a specific index. Those cases are mostly getting either the first or the last element.

I'd say that when getting the first element from a list type array, `_arr select 0` is clearer than `_arr params ["_firstElem"]` When using `params` on a list type array, it makes me think that it is a pseudo-object and the first element has some specific meaning.